presto-main/src/main/resources/webapp/src/components/SQLClient.jsx

```diff
@@ -0,0 +1,277 @@
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <div className="row">
+                <div className="col-xs-12">
+                    <Editor
+                        value={code}
+                        onValueChange={code => setCode(code)}
+                        highlight={code => highlight(code, languages.sql)}
+                        padding={10}
+                        style={{
+                            fontFamily: '"Fira code", "Fira Mono", monospace',
+                            fontSize: 14,
+                            borderWidth: '2px',
+                            borderColor: 'black',
+                            border: 'solid',
+                            margin: '4px',
+                        }}
+                        placeholder='Type your SQL query...'
+                    />
+                </div>
+            </div>
+        </>
+    );
+}
+
+function QueryResults({results}) {
+    const customStyles = {
+        headCells: {
+            style: {
+                padding: '2px', // override the cell padding for head cells
+                fontSize: '15px',
+            },
+        },
+        cells: {
+            style: {
+                padding: '2px', // override the cell padding for data cells
+                fontSize: '15px',
+            },
+        },
+    };
+
+    const getColumns = () => {
+        return results.columns.map( (row, index) => {
+            return {
+                name: row.name,
+                selector: row => row[index],
+            };
+        });
+    };
+
+    return (
+        <>
+            <div className="row">
+                <div className='col-xs-6'>
+                    <h3>Results</h3>
+                </div>
+                <div className="col-xs-6">
+                    <a style={{display: 'block', marginTop: '27px', marginBottom: '10px'}}
+                        href={"query.html?" + results.queryId} target="_blank"
+                        data-toggle="tooltip" data-trigger="hover" title="Query ID"
+                    >
+                        {results.queryId}
+                    </a>
+                </div>
+            </div>
+            <div className="row"><hr className="h3-hr"/></div>
+            {results.error && <div className="row"><div className="alert alert-danger" role="alert"><h4 className="text-center">{results.error.message}</h4></div></div>}
+            {results.data && <div className="row"><DataTable columns={getColumns()} data={results.data} theme='dark' customStyles={customStyles} striped='true' pagination /></div>}
+        </>
+    );
+}
+
+export default function SQLClientView() {
+    const [values, setValues] = React.useState({sql: '', running: false, results: undefined});
+
+    const executeSQL = (sqlInfo) => {
+        const client = createClient(sqlInfo.catalog, sqlInfo.schema);
```

Comment:
we're creating a new client each time we want to run a query, is this intented?

Reply:
Yes, I can change it to one instance and passing `catalog` and `schema` in each query. Which one you suggest?

Reply:
the way you did it is fine 👍